elearning-frontend/src/components/ProfileSettings.js

```diff
@@ -0,0 +1,228 @@
+    const avatarRef = useRef(null);
+    const [firstname, setFirstname] = useState('');
+    const [middlename, setMiddlename] = useState('');
+    const [lastname, setLastname] = useState('');
+    const [sex, setSex] = useState('');
+    const [birthdate, setBirthdate] = useState('');
+    const [phone, setPhone] = useState('');
+    const [address, setAddress] = useState('');
+    const [email, setEmail] = useState('');
+    const [username, setUsername] = useState('');
+    const [password, setPassword] = useState('');
+
+    const [loading, setLoading] = useState(true);
+    const usertoken = localStorage.getItem('user');
+
+    useEffect(() => {
+        if (!props.session) {
+            navigate('/')
+        }
+        else {
+            apiClient({
+                method: "post",
+                url: "/api/user",
+                headers: {
+                    Authorization: 'Bearer ' + usertoken
+                },
+                data: {
+                    token: usertoken
+                }
+            }).then(response => {
+                
+                if (response.data.user.profile.avatar !== null && response.data.user.profile.avatar != "") {
+                    setAvatar(response.data.user.profile.avatar);
+                    setProfilepic('http://127.0.0.1:8000/uploads/avatar/' + response.data.user.profile.avatar);
+                }
+
+                setFirstname(response.data.user.profile.firstname);
+                setMiddlename(response.data.user.profile.middlename);
+                setLastname(response.data.user.profile.lastname);
+                setSex(response.data.user.profile.sex);
+                setBirthdate(response.data.user.profile.birthdate);
+                setPhone(response.data.user.profile.phone);
+                setAddress(response.data.user.profile.address);
+                setEmail(response.data.user.email);
+                setUsername(response.data.user.username);
+                setPassword(response.data.user.password);
+
+                setLoading(false);
+            }).catch(error => {
+                console.log(error);
+            });
+        }
+    }, []);
+
+    const saveUser = (e) => {
+        e.preventDefault();
+        apiClient({
+            method: "post",
+            url: "/api/update-user",
+            headers: {
+                Authorization: 'Bearer ' + usertoken,
+                'Content-Type': 'multipart/form-data'
+            },
+            data: {
+                token: usertoken,
+                profilepic: profilepic,
+                avatar: avatar,
+                firstname: firstname,
+                middlename: middlename,
+                lastname: lastname,
+                sex: sex,
+                birthdate: birthdate,
+                phone: phone,
+                address: address,
+                email: email,
+                username: username,
+                password: password,
```

Comment:
this could be below since you have already put all your data in profile state then spread it out in data params
`data: {...profile}`